gymnasium/dev_wrappers/time_aware_observation.py

```diff
@@ -0,0 +1,140 @@
+                otherwise return time as remaining timesteps before truncation
+        """
+        super().__init__(env)
+        self.flatten = flatten
+        self.normalize_time = normalize_time
+        self.max_timesteps = self._get_max_timesteps(env)
+        self.num_envs = getattr(env, "num_envs", 1)
+
+        self.time_aware_observation_space = Dict(
+            obs=env.observation_space, time=Box(0, self.max_timesteps, (self.num_envs,))
+        )
+
+        if self.flatten:
+            self.observation_space = spaces.flatten_space(
+                self.time_aware_observation_space
+            )
+        else:
+            self.observation_space = self.time_aware_observation_space
+
+    def observation(self, observation: ObsType):
+        """Adds to the observation with the current time information.
+
+        Args:
+            observation: The observation to add the time step to
+
+        Returns:
+            The observation with the time information appended to
+        """
+        time_observation = self._get_time_observation()
+        observation = OrderedDict(obs=observation, time=time_observation)
+
+        return (
+            spaces.flatten(self.time_aware_observation_space, observation)
+            if self.flatten
+            else observation
+        )
+
+    def step(self, action: ActType):
+        """Steps through the environment, incrementing the time step.
+
+        In vectorized environments, if any of the environments
+        reach a terminal state, set the respective time value to zero.
+
+        Args:
+            action: The action to take
+
+        Returns:
+            The environment's step using the action.
+        """
+        self.timesteps += 1
+        observation, reward, terminated, truncated, info = super().step(action)
+
+        self.timesteps = jp.where(terminated, 0, self.timesteps)
+
+        return observation, reward, terminated, truncated, info
+
+    def reset(self, **kwargs):
+        """Reset the environment setting the time to zero.
+
+        Args:
+            **kwargs: Kwargs to apply to env.reset()
+
+        Returns:
+            The reset environment
+        """
+        self.timesteps = jp.zeros(self.num_envs)
+        return super().reset(**kwargs)
+
+    def _get_time_observation(self):
+        """Returns the `time` observation.
+
+        If normalization is applied, a value in the
+        range [0,1] is returned, otherwise, the
+        remaining timesteps before truncation.
+        """
+        return (
+            self.timesteps / self.max_timesteps
+            if self.normalize_time
+            else self.max_timesteps - self.timesteps
+        )
+
+    @staticmethod
+    def _get_max_timesteps(env: gymnasium.Env):
+        """Get the max episode steps before truncation occurs.
+
+        Args:
+            env: the provided environment
+        """
+        return (
+            env.get_attr("_max_episode_steps")[0]
```

Comment:
Reminder for future PRs: Do we want different environments to have different time limits? If so, we probably need to change this logic.